### PR TITLE
Feature/main app

### PR DESCRIPTION
**Pull request by hmightypirate**
Adding main functions to the app folder.

@eLeDe please check the missing function (drive_off) in motor.c. I just added "someting" in the commit but i am not sure if it is ok.

### CODE REVIEW

File: src/hal/clock.c [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 void sys_tick_handler(void) {
 
-  // TODO fill with cron calls
-
+  // FIXME: app dependent function
```
Comment (eLeDe) [inline src/hal/clock.c:5]: Remove this comment

**Review by eLeDe — CHANGES_REQUESTED** [review]
Remove comment in sys_tick routine

File: src/hal/clock.c (hunk 1 above)
Comment (hmightypirate) [reply src/hal/clock.c:5]: This comment was to indicate that we are using in hal a reference to a function in app, which is not the best approach. 

It is probably a best solution to move sys_tick_handler to some file in the app folder. What do you think?

Comment (punkto) [reply src/hal/clock.c:5]: We can address that in a future issue

File: src/main.c [hunk 2]
```diff
@@ -1,4 +1,6 @@
 #include "setup.h"
+#include "mouse_app.h"
+#include "mmlib/speed.h"
 
 /**
  * @brief Empty main
```
Comment (punkto) [inline src/main.c:6]:
```suggestion
 * @brief Program entry point
```

**Pull request merged**